ledger/participant-integration-api/src/main/scala/platform/store/backend/StorageBackend.scala

```diff
@@ -9,12 +9,51 @@ import com.daml.ledger.participant.state.v1.Offset
 import com.daml.platform.store.DbType
 import com.daml.platform.store.backend.postgresql.PostgresStorageBackend
 
-// TODO append-only: add detailed scaladoc
+/** Encapsulates the interface which hides database technology specific implementations for parallel ingestion.
+  *
+  * @tparam DB_BATCH Since parallel ingestion comes also with batching, this implementation specific type allows separation of the CPU intensive batching operation from the pure IO intensive insertBatch operation.
+  */
 trait StorageBackend[DB_BATCH] {
+
+  /** The CPU intensive batching operation hides the batching logic, and the mapping to the database specific representation of the inserted data.
+    * This should be pure CPU logic without IO.
+    *
+    * @param dbDtos is a collection of DBDTOV1 which from the batch is formed
+    * @return the database-specific batch DTO, which can be inserted via insertBatch
+    */
   def batch(dbDtos: Vector[DBDTOV1]): DB_BATCH
+
+  /** Using a JDBC connection, a batch will be inserted into the database.
+    * No significant CPU load, mostly blocking JDBC communication with the database backend.
+    *
+    * @param connection to be used for inserting a batch
+    * @param batch to be inserted
+    */
   def insertBatch(connection: Connection, batch: DB_BATCH): Unit
+
+  /** This method is used to update the parameters table: setting the new observable ledger-end, and other parameters.
+    * No significant CPU load, mostly blocking JDBC communication with the database backend.
+    *
+    * @param connection to be used for upating the parameters table
+    * @param params the actual parameters
+    */
   def updateParams(connection: Connection, params: StorageBackend.Params): Unit
+
+  /** Custom initialization code before the start of ingestion.
+    * This method is responsible to recover after a possibly non-graceful stop of previous indexing.
+    * No significant CPU load, mostly blocking JDBC communication with the database backend.
+    *
+    * @param connection to be used for initialization
+    * @return the actual LedgerEnd, which should be the basis for further indexing.
+    */
   def initialize(connection: Connection): StorageBackend.LedgerEnd
+
+  /** Query the actual ledger-end, read from the parameters table.
```

Comment:
```suggestion
  /** Query the actual ledgerEnd, read from the parameters table.
```
Same comment as above above about the "actual"